src/lib.rs

```diff
@@ -17,6 +23,36 @@ pub fn count_syllables(word: &str) -> u8 {
     count
 }
 
+pub fn read_dir(dir: String) -> HashMap<String, Vec<String>> {
+    let paths = fs::read_dir(dir).unwrap();
+    let mut map: HashMap<String, Vec<String>> = HashMap::new();
+    for path in paths {
+        let mut cur: Vec<String> = Vec::new();
+        let curpath = path.unwrap().path();
+        let filename = String::from(curpath.file_name().unwrap().to_str().unwrap());
+        let filetype = String::from(&filename[..(filename.len() - 4)]);
```

Comment:
```suggestion
        let filename = curpath.file_name().unwrap().to_str().unwrap().to_string();
        let filetype = filename[..(filename.len() - 4)].to_string();
```

Reply:
i just want to keep using to `to_string()` for consistency

Reply:
Changed